src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -143,6 +166,98 @@ def signal_handler(sig, frame):
     raise KeyboardInterrupt
 
 
+def log_stack_recursively(task: asyncio.Task, elapse_time: float):
+    """Recursively log the frame of a task or coroutine.
+    Traditional stacktrace would stop at the first awaited nested inside the coroutine.
+
+    :param task: Task to log
+    :type task_or_coroutine: asyncio.Task
+    :param elapse_time: Seconds elapsed since the task started
+    :type elapse_time: float
+    """
+    # We cannot use task.get_stack() to get the stack, because only one stack frame is
+    # returned for a suspended coroutine because of the implementation of CPython
+    # Ref: https://github.com/python/cpython/blob/main/Lib/asyncio/tasks.py
+    # "only one stack frame is returned for a suspended coroutine."
+    task_or_coroutine = task
+    frame_summaries = []
+    while True:
+        if isinstance(task_or_coroutine, asyncio.Task):
+            # For a task, get the coroutine it's running
```

Comment:
Maybe we can have more comments here?
Seems there are two procedure:
1 extract_frame_summaries_in_stack
2 get_msg_from_frame_summaries

Reply:
Refactored and added comment.